### PR TITLE
Gfortran fixes

### PR DESCRIPTION
**Pull request by JamieJQuinn**
This PR fixes a number of segfaults which occurred only when running trove built with gfortran.

One difference between  the intel and gfortran compilers is in the evaluation of compound logical expressions in if statements, for example `if( present(array) .and. any(array == 0.0) )`. The intel compiler will happily evaluate this if `array` is not present or not allocated; gfortran will not. This error unfortunately appears as a runtime error so isn't particularly easy to diagnose. 

The other main difference was in how gfortran treats reading and writing values. Intel will happily convert between int and logical when reading and writing; gfortran will not. 

I have also added a check in the makefile to determine which gfortran version is being run to optionally add the new (in gcc 10) flag `-fallow-argument-mismatch`.

### CODE REVIEW

File: fields.f90 [hunk 1]
```diff
@@ -1561,6 +1561,11 @@ subroutine FLReadInput(NPTorder,Npolyads,Natoms,Nmodes,Jrot)
             !
          endif 
          !
+         if (.not. symmetry_defined) then
```
Comment (JamieJQuinn) [inline fields.f90:1564]: @Trovemaster Sergey, this is my solution to the diagonalizer-symmetry ordering issue. The if statement below was the one causing the segfault; `job%nroots` was the array being accessed without it first being allocated in the code handling the symmetry input block.

Comment (Trovemaster) [reply fields.f90:1564]: Thanks! This makes sense.

**Comment by JamieJQuinn** [comment]
@Trovemaster Glad you're happy with the symmetry fix. Are you happy for the rest of this PR to be merged?

**Comment by Trovemaster** [comment]
Yes. The changes makes sense. Thanks!

**Pull request merged**